envoy.code.check/envoy/code/check/abstract/glint.py

```diff
@@ -0,0 +1,139 @@
+
+import asyncio
+import re
+from functools import cached_property
+from typing import Dict, Iterator, List, Set, Tuple
+
+import abstracts
+
+from aio.core import dev
+from aio.core.functional import async_map, async_property, async_set
+
+from envoy.base import utils
+from envoy.code.check import abstract
+
+
+NOGLINT_RE = (
+    r"[\w/]*\.patch$",
+    r"^test/[\w/]*_corpus/[\w/]*",
+    r"^tools/[\w/]*_corpus/[\w/]*",
+    r"[\w/]*password_protected_password.txt$")
+
+
+class AGlintCheck(abstract.ACodeCheck, metaclass=abstracts.Abstraction):
+
+    @classmethod
+    def have_newlines(cls, paths: str) -> Set[Tuple[str, bool]]:
+        return set(
+            (target,
+             (utils.last_n_bytes_of(target)
+              == b'\n'))
+            for target in paths)
+
+    @classmethod
+    def filter_files(cls, files, match) -> Set[str]:
+        return set(
+            path
+            for path
+            in files
+            if not match(path))
+
+    @async_property
+    async def checker_files(self) -> Set[str]:
+        return self.filter_files(
+            await self.directory.files,
+            self.noglint_re.match)
+
+    @async_property
+    async def files_with_mixed_tabs(self) -> Set[str]:
+        files = await self.files_with_preceeding_tabs
+        return (
+            await self.directory.grep(["-lP", r"^ ", *files])
+            if files
+            else set())
+
+    @async_property
+    async def files_with_preceeding_tabs(self) -> Set[str]:
+        files = await self.files
+        if not files:
+            return set()
+        # This is a workaround for a blocking issue sending
+        # very large cli args to a subprocess.
+        # TODO: generalize this solution in directory.grep
+        if len(files) < 1000:
+            return await self.directory.grep(
+                ["-lP", r"^\t",
+                 *await self.files])
+        return (
```

Comment:
these lines are not currently covered, but i want to move this to aio.core.directory - so will shift and address tests in #352